src/reporting/lambda/custom-resource/quicksight/index.ts

```diff
@@ -170,9 +172,16 @@ const _onUpdate = async (quickSight: QuickSight, awsAccountId: string, principal
     const dashboardDefProps: QuickSightDashboardDefProps = props.dashboardDefProps;
     logger.info('dashboardDefProps', JSON.stringify(dashboardDefProps));
 
+    let newPrincipalArn = undefined;
+    if (oldProps.quickSightUser !== props.quickSightUser) {
```

Comment:
Add test for covering this case where the principal is changed when updating the template

Reply:
already added.  <Update QuickSight dashboard - permission update>